app/controllers/campaigns_controller.rb

```diff
@@ -50,6 +50,30 @@ def associate_seller_distributor
     json_response(campaign_json)
   end
 
+  # POST /campaigns/:id/gift_card
+  def generate_campaign_gift_cards
+    request_params = generate_campaign_gift_cards_params
+    unless @campaign.amount_raised > @campaign.amount_allocated + request_params['gift_card_amount_cents']
+      raise InvalidLineItem, "Request amount exceeds unallocated amount in campaign. Unallocated amount: #{
+        @campaign.amount_raised - @campaign.amount_allocated
+      }"
+    end
+
+    WebhookManager::GiftCardCreator.call(
```

Comment:
not sure if we wanted to send out receipts for this as well?

Reply:
I don't think we wanna send out receipts

Reply:
got it